### PR TITLE
Add helper tool for making sure build, lint, and test are run through Turborepo

### LINKED ISSUE #33459: Block `build`, `lint`, and `test` Scripts Without `turbo`
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Right now it's possible to run `pnpm build` for a plugin, package, or tool. This has a negative impact, however, since it bypasses the `turbo` cache and may leave dependencies unbuilt. We should look into detecting `turbo` execution and do nothing if the commands are not ran using it. The first thing that comes to mind would be looking for an environment variable and seeing if one is set. If not, perhaps we need to upstream a change to Turborepo to add one.

### PR DESCRIPTION
**Pull request by jonathansadowski**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a helper tool for verifying that `build`, `lint`, and `test` scripts cannot be directly run in any of the plugins.

Note: Because of the `prepare` scripts for the `e2e-*` packages, which attempt to build, I wasn't able to apply this treatment to them.

Closes #33459.

### How to test the changes in this Pull Request:

1. `pnpm install`
2. Attempt to build a project directly with `pnpm`, such as: `cd plugins/woocommerce && pnpm run build`
3. Verify that you're given an error message and that the error message gives you sensible instructions for actually building the project with Turborepo.
4. Repeat the previous steps with `lint` and `test` for any projects that have those scripts defined.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Comment by botwoo** [comment]
:bar_chart: Test reports for this pull request have been published and are accessible through the following links:

- [REST API Test Report](https://woocommerce.github.io/woocommerce-test-reports/pr/33573/api/)
- E2E Test Report (TBC)

Latest commit referenced in the reports: `Update node version for require-turbo` 1d7c9495a3f9688e5871a0c6b9e2def0b36e83e3
This comment will automatically be updated with the latest referenced commit when you push new changes to this pull request.

---

_Visit the [WooCommerce Test Reports](https://woocommerce.github.io/woocommerce-test-reports) homepage to view all published reports. See the [FAQs](https://github.com/woocommerce/woocommerce-test-reports/wiki/FAQs) page if you're having problems accessing them._

File: tools/require-turbo/package.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"name": "require-turbo",
+	"version": "0.0.0",
+	"description": "A tool for checking that package scripts are run through Turborepo",
+	"main": "index.js",
+	"author": "Automattic",
+	"license": "GPL-2.0-only",
+	"engines": {
+		"node": ">=12.0.0"
```
Comment (roykho) [inline tools/require-turbo/package.json]: Since we've updated the node versions in other configs to `^16.13.1` perhaps we make this consistent.

Comment (jonathansadowski) [reply tools/require-turbo/package.json]: Good call, updated.

**Review by roykho — APPROVED** [review]
LGTM! Tested well. Good job!

**Pull request merged**